kombu/transport/redis.py

```diff
@@ -382,7 +382,9 @@ def ack(self, delivery_tag):
     def reject(self, delivery_tag, requeue=False):
         if requeue:
             self.restore_by_tag(delivery_tag, leftmost=True)
-        self.ack(delivery_tag)
+        else:
```

Comment:
it would be great if you could add unit test for this change and if possible, integration test which will give us a lot of confidence